routemaster/state_machine/actions.py

```diff
@@ -100,15 +103,29 @@ def process_retries(
             conn,
         )
 
-        for label_name, metadata in relevant_labels.items():
-            label = LabelRef(name=label_name, state_machine=state_machine.name)
+    for label_name, metadata in relevant_labels.items():
+        label = LabelRef(name=label_name, state_machine=state_machine.name)
+        could_progress = False
 
-            with conn.begin():
-                _process_action_with_metadata(
+        with app.db.begin() as conn:
+            try:
+                lock_label(label, conn)
+                current_state = get_current_state(label, state_machine, conn)
```

Comment:
Do we need to explicitly check whether the label has been deleted here? (I can't remember whether we move deleted labels to an 'exit' state; if we do, then there's no need)

Reply:
We do move them to `None`.